### ISSUE TITLE
wallet tool salvage always fails with "Assertion `new_tx' failed"

### ISSUE THREAD

**Issue by MarcoFalke** [bitcoin/bitcoin#19079]
Error:

```
bitcoin-wallet: wallet/walletdb.cpp:291: auto ReadKeyValue(CWallet *, CDataStream &, CDataStream &, CWalletScanState &, std::string &, std::string &)::(anonymous class)::operator()(CWalletTx &, bool) const: Assertion `new_tx' failed.
```

Steps to reproduce: #19078

**Comment by vahidjalaliii** [comment]
> خطا:
> 
> ```
> bitcoin-wallet: wallet/walletdb.cpp:291: auto ReadKeyValue(CWallet *, CDataStream &, CDataStream &, CWalletScanState &, std::string &, std::string &)::(anonymous class)::operator()(CWalletTx &, bool) const: Assertion `new_tx' failed.
> ```
> 
> مراحل تولید مثل: [# 19078](https://github.com/bitcoin/bitcoin/pull/19078)

۱۹۰۷۹

**Comment by achow101** [comment]
I believe this can be closed as #19805 sidesteps the problem.
